src/Symfony/Component/Validator/Tests/Constraints/SemVerValidatorTest.php

```diff
@@ -0,0 +1,363 @@
+
+        $this->buildViolation('Custom maximum version message')
+            ->setParameter('{{ value }}', '"2.0.0"')
+            ->setParameter('{{ max }}', '1.0.0')
+            ->setCode(SemVer::TOO_HIGH_ERROR)
+            ->assertRaised();
+    }
+
+    public static function getValidLooseSemVersions(): iterable
+    {
+        // Full versions
+        yield ['0.0.0'];
+        yield ['1.0.0'];
+        yield ['1.2.3'];
+        yield ['10.20.30'];
+
+        // Partial versions
+        yield ['1'];
+        yield ['1.2'];
+        yield ['10.20'];
+
+        // With prefix
+        yield ['v1.0.0'];
+        yield ['v1.2.3'];
+        yield ['v1'];
+        yield ['v1.2'];
+
+        // With pre-release
+        yield ['1.0.0-alpha'];
+        yield ['1.0.0-alpha.1'];
+        yield ['1.0.0-0.3.7'];
+        yield ['1.0.0-x.7.z.92'];
+        yield ['1.0.0-alpha+001'];
+        yield ['1.0.0+20130313144700'];
+        yield ['1.0.0-beta+exp.sha.5114f85'];
+        yield ['1.0.0+21AF26D3----117B344092BD'];
+
+        // Complex examples
+        yield ['1.2.3-alpha.1.2+build.123'];
+        yield ['v1.2.3-rc.1+build.123'];
+    }
+
+    public static function getValidStrictSemVersions(): iterable
+    {
+        // Only valid according to official SemVer spec (no v prefix)
+        yield ['0.0.0'];
+        yield ['1.0.0'];
+        yield ['1.2.3'];
+        yield ['10.20.30'];
+        
+        // With pre-release
+        yield ['1.0.0-alpha'];
+        yield ['1.0.0-alpha.1'];
+        yield ['1.0.0-0.3.7'];
+        yield ['1.0.0-x.7.z.92'];
+        
+        // With build metadata
+        yield ['1.0.0+20130313144700'];
+        yield ['1.0.0+21AF26D3----117B344092BD'];
+        
+        // With both
+        yield ['1.0.0-alpha+001'];
+        yield ['1.0.0-beta+exp.sha.5114f85'];
+        yield ['1.2.3-alpha.1.2+build.123'];
+    }
+
+    public static function getInvalidSemVersions(): iterable
+    {
+        yield ['v'];
+        yield ['1.2.3.4'];
+        yield ['01.2.3'];
+        yield ['1.02.3'];
+        yield ['1.2.03'];
+        yield ['1.2-alpha'];
+        yield ['1.2.3-'];
+        yield ['1.2.3-+'];
+        yield ['1.2.3-+123'];
+        yield ['1.2.3-'];
+        yield ['+invalid'];
+        yield ['-invalid'];
+        yield ['-invalid+invalid'];
+        yield ['-invalid.01'];
+        yield ['alpha'];
+        yield ['1.2.3.DEV'];
+        yield ['1.2-SNAPSHOT'];
+        yield ['1.2.31.2.3----RC-SNAPSHOT.12.09.1--..12+788'];
+        yield ['1.2-RC-SNAPSHOT'];
+        yield ['1.0.0+'];
+        yield ['1.0.0-'];
+    }
+
+    public static function getInvalidStrictSemVersions(): iterable
+    {
+        // Versions with v prefix (not allowed in strict mode)
+        yield ['v1.0.0'];
+        yield ['v1.2.3'];
+        yield ['v1.0.0-alpha'];
+        yield ['v1.0.0+20130313144700'];
+        
+        // Partial versions (not allowed in strict mode)
```

Comment:
```suggestion
        // Partial versions
```

(Self explanatory in the data provider name)